docrane/container.py

```diff
@@ -65,6 +73,19 @@ def ensure_running(self, force_restart=False):
                     LOG.warning("Image ID has changed. Restarting container.")
                 break
 
+        try:
+            image_found = util.pull_image(
+                self.docker_params.get('image'), self.docker_params.get('tag'))
+            assert image_found
+        except AssertionError:
+            # The image wasn't found
+            LOG.warning("Couldn't find image %s:%s. Delaying next scan.. " % (
+                self.docker_params.get('image'), self.docker_params.get('tag')))
+            self.delay = 4
```

Comment:
removing the delay would simplify a lot of the container state maintenance code.